### PR TITLE
Added test framework and initial tests

### PR DESCRIPTION
**Pull request by jmkulwik**
Brings us to ~30% code coverage

There are three classes that don't yet have coverage that definitely need coverage: `UserCodeCompiler`, `UserProjectFileManager`, and `JavaRunner`. There is a follow-up task to set up tests for these classes: https://codedotorg.atlassian.net/browse/CSA-237

image

### CODE REVIEW

File: org-code-javabuilder/lib/src/test/java/org/code/javabuilder/AWSInputAdapterTest.java
Comment (jamescodeorg) [inline org-code-javabuilder/lib/src/test/java/org/code/javabuilder/AWSInputAdapterTest.java:42]: This verification looks unrelated the main test?  (I see that it has its own test in deletesAllReadMessages() below.)

File: org-code-javabuilder/lib/src/test/java/org/code/javabuilder/AWSInputAdapterTest.java
Comment (jamescodeorg) [inline org-code-javabuilder/lib/src/test/java/org/code/javabuilder/AWSInputAdapterTest.java:34]: Some of this setup looks test-specific.  I recommend moving anything that is test-specific out to the individual test themselves, adding helper functions to reduce code duplication if needed.

File: org-code-javabuilder/lib/src/test/java/org/code/javabuilder/AWSInputAdapterTest.java
Comment (jamescodeorg) [inline org-code-javabuilder/lib/src/test/java/org/code/javabuilder/AWSInputAdapterTest.java:41]: As an example, the setup logic that adds a "hello" message should be in this test method (and not in setUp) because the test logic depends on the message body being "hello".

File: org-code-javabuilder/lib/src/test/java/org/code/javabuilder/InputRedirectionStreamTest.java
Comment (jamescodeorg) [inline org-code-javabuilder/lib/src/test/java/org/code/javabuilder/InputRedirectionStreamTest.java:18]: As another example, I would consider moving the setup code here into each function (or create a helper) so the arrange-act-assert pattern in each unit test clear.  Alternatively, you can instantiate `stream` and `inputAdapter` here in setup but at least the `when` line should go in each test (and be tailored to be only what is needed by the test).

File: org-code-javabuilder/lib/src/test/java/org/code/javabuilder/InputRedirectionStreamTest.java
Comment (jamescodeorg) [inline org-code-javabuilder/lib/src/test/java/org/code/javabuilder/InputRedirectionStreamTest.java:15]: Heh, one of the major decisions that you have to make when writing the first unit tests -- is "setup" one word or two?  (It's spelled "setup" here but "setUp" above.)

**Review by jamescodeorg — COMMENTED** [review]
Sorry for the delay, I'm still catching up on code reviews.

**Comment by jmkulwik** [comment]
@jamescodeorg Thank you for the review! Some other folks are starting to do dev work on Javabuilder - I'm going to merge the portions of this that have no comments in order to unblock test writing on in-process work.

**Comment by jamescodeorg** [comment]
Yes, please feel free to merge some or all of this change.  I know this is all work-in-progress so we don't need to get things perfect before merging.

File: org-code-javabuilder/lib/src/test/java/org/code/javabuilder/AWSInputAdapterTest.java
Comment (jmkulwik) [reply org-code-javabuilder/lib/src/test/java/org/code/javabuilder/AWSInputAdapterTest.java]: Fixed!

File: org-code-javabuilder/lib/src/test/java/org/code/javabuilder/AWSInputAdapterTest.java
Comment (jmkulwik) [reply org-code-javabuilder/lib/src/test/java/org/code/javabuilder/AWSInputAdapterTest.java]: Thanks! That cleaned things up a bunch!

File: org-code-javabuilder/lib/src/test/java/org/code/javabuilder/AWSInputAdapterTest.java
Comment (jmkulwik) [reply org-code-javabuilder/lib/src/test/java/org/code/javabuilder/AWSInputAdapterTest.java:42]: Fixed!

File: org-code-javabuilder/lib/src/test/java/org/code/javabuilder/InputRedirectionStreamTest.java
Comment (jmkulwik) [reply org-code-javabuilder/lib/src/test/java/org/code/javabuilder/InputRedirectionStreamTest.java]: Thanks! That cleaned things up a bunch!

File: org-code-javabuilder/lib/src/test/java/org/code/javabuilder/InputRedirectionStreamTest.java
Comment (jmkulwik) [reply org-code-javabuilder/lib/src/test/java/org/code/javabuilder/InputRedirectionStreamTest.java]: Haha ok. I standardized :)

File: org-code-javabuilder/lib/src/test/java/org/code/javabuilder/AWSInputAdapterTest.java
Comment (jamescodeorg) [inline org-code-javabuilder/lib/src/test/java/org/code/javabuilder/AWSInputAdapterTest.java:26]: (minor) A header comment for this method might be useful.  Also, since you are setting precedents here, do you generally want to put helper functions before or after the tests?

**Review by jamescodeorg — APPROVED** [review]

**Review by molly-moen — APPROVED** [review]
Just a couple places I could use clarifying comments, otherwise looks good!

File: org-code-javabuilder/lib/src/test/java/org/code/javabuilder/AWSInputAdapterTest.java
Comment (jmkulwik) [reply org-code-javabuilder/lib/src/test/java/org/code/javabuilder/AWSInputAdapterTest.java:26]: I'm used to putting them before because that's generally what our JS code does. Looking at our Ruby code, it has helpers more rarely, but when they are there, there seems to be a bit of a mix of before and after, possibly more after. Any reasons to favor one way over the other? I think I prefer before so I don't make any assumptions about built-in methods when I'm reading the code, but it's not a strong preference.

Comment (jamescodeorg) [reply org-code-javabuilder/lib/src/test/java/org/code/javabuilder/AWSInputAdapterTest.java:31]: I think I'm more used to seeing helper methods after the tests but don't know if that's a best practice or just a quirk of codebases I've worked on in the past.  As the person who's blazing the trail here, you get to choose!  :-)

**Pull request merged**
